src/pages/workspaces/workspace/jobHistory/CallCacheWizard.js

```diff
@@ -0,0 +1,270 @@
+import _ from 'lodash/fp'
+import { Fragment, useState } from 'react'
+import { div, h, hr } from 'react-hyperscript-helpers'
+import ReactJson from 'react-json-view'
+import Select from 'react-select'
+import { ButtonPrimary, ButtonSecondary } from 'src/components/common'
+import ErrorView from 'src/components/ErrorView'
+import { icon } from 'src/components/icons'
+import { TextInput } from 'src/components/input'
+import { breadcrumbHistoryCaret } from 'src/components/job-common'
+import Modal from 'src/components/Modal'
+import { Ajax } from 'src/libs/ajax'
+import * as Style from 'src/libs/style'
+import * as Utils from 'src/libs/utils'
+
+
+const CallCacheWizard = ({
+  onDismiss, workflowId, callFqn, attempt, index
+}) => {
+  /*
+   * State setup
+   */
+
+  const [otherWorkflowId, setOtherWorkflowId] = useState()
+  const [otherWorkflowMetadata, setOtherWorkflowMetadata] = useState()
+  const [otherCallFqn, setOtherCallFqn] = useState()
+  const [otherIndex, setOtherIndex] = useState()
+  const [otherAttemptSucceeded, setOtherAttemptSucceeded] = useState()
+  const [otherCallSelected, setOtherCallSelected] = useState(false)
+  const [diff, setDiff] = useState()
+  const [metadataFetchError, setMetadataFetchError] = useState()
+  const [diffError, setDiffError] = useState()
+
+  const signal = Utils.useCancellation()
+
+  /*
+   * Data Fetchers
+   */
+
+  const readCalls = async otherWf => {
+    try {
+      const includeKey = [
+        'end', 'start', 'executionStatus'
+      ]
+      const excludeKey = []
+      const wf = await Ajax(signal).CromIAM.workflowMetadata(otherWf, includeKey, excludeKey)
+      setOtherWorkflowMetadata(wf)
+    } catch (error) {
+      console.log(error)
+      if (error instanceof Response) setMetadataFetchError(await error.text())
+      else if (_.isObject(error)) setMetadataFetchError(JSON.stringify(error))
+      else setMetadataFetchError(error)
+    }
+  }
+
+  const fetchDiff = async (otherWf, otherCall, otherIx) => {
+    try {
+      const diff = await Ajax(signal).CromIAM.callCacheDiff(workflowId, callFqn, Number(index), otherWf, otherCall, Number(otherIx))
+      setDiff(diff)
+    } catch (error) {
+      console.log(error)
+      if (error instanceof Response) setDiffError(await error.text())
+      else if (_.isObject(error)) setDiffError(JSON.stringify(error))
+      else setDiffError(error)
```

Comment:
There are two calls made from the wizard at different stages - 

1. Fetching the "other" workflow's metadata so that we can list calls in the drop down
2. Fetching the call cache diff between two workflows

Either of them can fail, independently, for a variety of different reasons. The most common failure in (1) is invalid workflow ID or a workflow which the user doesn't have access to. A timeout is also possible. For (2) there are also a couple of edge cases which can cause the endpoint to give us an error instead of a correct response. Eg this example from the PR description:

* Comparing the diff between two Dev workflows (currently "demonstrates" error handling because of a Cromwell bug):
http://localhost:3000/#workspaces/general-dev-billing-account/scale-abort-test/job_history/c4752583-839c-4319-bb0d-5eae15fa1fdc/d471802e-e1cc-405c-a816-536a6a7a69db
http://localhost:3000/#workspaces/general-dev-billing-account/scale-abort-test/job_history/b6647bde-603d-43d4-a24f-3b68bf04cf0e/2520d817-83ee-4e4d-961f-e582d29b1b6c